wscript

```diff
@@ -158,8 +158,12 @@ main_dependencies = [
         'func': compose_checks(
             check_statement('android/api-level.h', '(void)__ANDROID__'),  # arbitrary android-specific header
             check_cc(lib="android"),
-            check_cc(lib="EGL"),
         )
+    }, {
+        'name': '--android-jni',
+        'desc': 'Android environment (JNI)',
+        'deps': 'android',
+        'func': check_cc(lib="EGL"),
```

Comment:
a check for EGL seems kinda out of place here

Reply:
It was not added in this PR though. Also note that `egl-helpers` depends on it.

Reply:
The dependency in `egl-helpers` is from https://github.com/mpv-player/mpv/commit/8fc21fd0d5aa733eb4388c68d22c8f1748f16e33.

Reply:
Maybe better to call it `android-egl` instead?

I assume this works for you because your compilation environment in termux is missing both jni and egl headers?

Reply:
@tmm1 The EGL library is not the missing part in Termux, but JVM/JNI. mpv is not used as a library from the Java layer there. That's why I also include `vo_mediacodec_embed.c` into `android-jni` even when we can compile with it in `android`, as to actually use it, you still need to deal with the Surface stuff from the Java layer.

Also if your AudioTrack PR is going to be merged in the future, it belongs to `android-jni` as well. So I think the naming is fine. (I don't think that was @sfan5's concern anyway.)

Reply:
I understand the intent here, but an `android-jni` flag which depends only on `libEGL` doesn't make much sense.

It would be better if this was an automatic check (i.e. `name: android-jni` instead of `name: --android-jni`), and it depended on something which could be detected and enabled automatically.

Does your compilation environment include jni.h?

Reply:
That would be best, but since `--disable-android-jni` is fine enough for us I am sort of neutral to that. The thing is I don't know how we can check "we are going to have JNIEnv and friends where the build is used". (Doesn't sound like possible either.)

And sure it does. We use the NDK as well (and its headers when build on-site).